airflow/providers/http/hooks/http.py

```diff
@@ -54,6 +60,10 @@ def __init__(
         self._retry_obj: Callable[..., Any]
         self.auth_type: Any = auth_type
 
+    @cached_property
+    def conn(self):
+        return self.get_connection(self.http_conn_id)
```

Comment:
This is a bit confusing because get_conn returns Request.Session, so this property should return that as well. We have a similar situation with AwsBaseHook.
https://github.com/apache/airflow/blob/0cd3b11f3a5c406fbbd4433d8e44d326086db634/airflow/providers/amazon/aws/hooks/base_aws.py#L492-L493
https://github.com/apache/airflow/blob/0cd3b11f3a5c406fbbd4433d8e44d326086db634/airflow/providers/amazon/aws/hooks/base_aws.py#L508
What can we do to make this code more consistent?

Reply:
Good point. I can see where it is misleading. There is a little wrinkle with session headers that can change between calls but I'll dig in.

Reply:
Although is it less confusing if the property was called `connection` instead?

Reply:
I am wondering if this cache should not be part of BaseHook as a similar problem will also occur in user code. WDYT?

Reply:
A little timid with that impactful of a change, but I knew there was an immediate caching benefit in `HttpHook`. Selfishly I suppose.

Reply:
Oh I remember, the issue was figuring out how to get around the connection backend priority if there were multiple connections with the same `conn_id` but set with OS variables, database, etc. and an adjacent thought on what happens if the same `conn_id` exists for multiple `conn_types`?

Reply:
We can break it down into two changes. One that will be released in providers packages and fix the problem locally and the other change that will require a new version of Airflow but fix the problem globally.